Objective-C/CBLConsoleLogger.m

```diff
@@ -45,7 +45,8 @@ - (void) logWithLevel: (CBLLogLevel)level domain: (CBLLogDomain)domain message:
     
     NSString* levelName = CBLLog_GetLevelName(level);
     NSString* domainName = CBLLog_GetDomainName(domain);
-    NSLog(@"CouchbaseLite %@ %@: %@", domainName, levelName, message);
+    os_log_t log = os_log_create("CouchbaseLite", "OSDebug");
+    os_log(log, "CouchbaseLite %@ %@: %@", domainName, levelName, message);
```

Comment:
There are different types of os_log function based on the log level. I think we should use those functions. What is .NET using?

Reply:
.NET is mapping what we have to os_log as
```new Dictionary<LogLevel, CoreFoundation.OSLogLevel>
            {
                [LogLevel.Debug] = CoreFoundation.OSLogLevel.Debug,
                [LogLevel.Verbose] = CoreFoundation.OSLogLevel.Info, // No verbose level in Apple
                [LogLevel.Info] = CoreFoundation.OSLogLevel.Info,
                [LogLevel.Warning] = CoreFoundation.OSLogLevel.Error, // No warning level in Apple
                [LogLevel.Error] = CoreFoundation.OSLogLevel.Error,
            };
  ```          
  
My understanding is that these NSLog are only informative and have nothing to do with the actual CBL Logging, thus a similar change with .NET is not needed